Mathlib/Analysis/Complex/Hadamard.lean

```diff
@@ -0,0 +1,487 @@
+    rw [hpos]
+    refine DifferentiableAt.mul ?_ ?_
+    · apply DifferentiableAt.const_cpow (DifferentiableAt.sub_const (differentiableAt_id') 1) _
+      left; simp only [Ne.def, ofReal_eq_zero]; rwa [eq_comm]
+    · refine DifferentiableAt.const_cpow ?_ ?_
+      · rw [differentiableAt_neg_iff]
+        apply differentiableAt_id'
+      · left; simp only [Ne.def, ofReal_eq_zero]; rwa [eq_comm]
+
+
+
+lemma diffContOnCl_F' (f : ℂ → ℂ) (n : ℕ) (hd : DiffContOnCl ℂ f (strip 0 1)) :
+    DiffContOnCl ℂ (F' n f) (strip 0 1) := by
+  refine DiffContOnCl.smul (DiffContOnCl.smul hd (diffContOnCl_invInterpStrip f) )
+    <| Differentiable.diffContOnCl <| Differentiable.cexp <| Differentiable.mul ?_
+      (differentiable_const _)
+  rw [differentiable_sub_const_iff]
+  simp only [cpow_two, differentiable_id', Differentiable.pow]
+
+
+
+-- The function `f` is bounded by `sSupAbsIm`
+lemma abs_le_sSupAbsIm (f : ℂ → ℂ) (z : ℂ) (hD : z ∈ (closedStrip 0 1))
+    (hB : BddAbove ((abs ∘ f) '' (closedStrip 0 1))) : abs (f z) ≤ sSupAbsIm f (z.re) := by
+  refine le_csSup ?_ ?_
+  · apply BddAbove.mono (image_subset (abs ∘ f) _) hB
+    apply preimage_mono
+    simp only [singleton_subset_iff]
+    exact hD
+  · apply mem_image_of_mem (abs ∘ f)
+    simp only [mem_preimage, mem_singleton]
+
+lemma cocompact_strip.basis : HasBasis (cocompactStrip 0 1) (fun _ : ℝ × ℝ ↦ True )
+    (fun T ↦ (im ⁻¹' (Iic (T.1) ∪ Ici (T.2)) ∩ re ⁻¹' Icc 0 1)) := by
+  simp only [cocompactStrip, comap_principal, Real.cocompact_eq]
+  rw [← true_and True]
+  exact HasBasis.inf_principal (HasBasis.comap _ (HasBasis.sup atBot_basis atTop_basis)) _
+
+lemma cocompact_strip.mem_sets {s : Set ℂ} : s ∈ cocompactStrip 0 1 ↔ ∃ T, ∀ (z : ℂ),
+    z.re ∈ Icc 0 (1 : ℝ) → T ≤ |z.im| → z ∈ s := by
+  rw [HasBasis.mem_iff cocompact_strip.basis]
+  constructor
+  · rintro ⟨⟨a,b⟩, h⟩
+    simp only [preimage_union, ge_iff_le, zero_le_one, not_true, gt_iff_lt, true_and] at h
+    use max (-a) b
+    intros z hset him
+    apply h
+    refine mem_inter ?_ hset
+    simp only [mem_preimage, mem_Iic, mem_Ici, mem_union]
+    simp only [abs_ofReal, max_le_iff] at him
+    cases' him with hima himb
+    by_cases 0 ≤ z.im
+    · right; rwa [← _root_.abs_of_nonneg h]
+    · left; rwa [← neg_le_neg_iff, ← abs_of_neg (not_le.mp h)]
+  · rintro ⟨T, h⟩
+    use (-T, T)
+    simp only [preimage_union, true_and]
+    rintro z ⟨him, hset⟩
+    apply h _ hset
+    rwa [le_abs, ← le_neg, or_comm, ← mem_Iic, ← mem_Ici,
+      ← mem_union, ← mem_preimage, preimage_union]
+
+lemma _root_.Real.tendsto_sq_cocompact_atTop : Tendsto (fun x ↦ x^2) (cocompact ℝ) atTop := by
+  convert_to Tendsto (fun x ↦ ‖x‖ * ‖x‖) (cocompact ℝ) atTop
+  · ext x; rw [pow_two]; exact (abs_mul_abs_self _).symm
+  · exact tendsto_norm_cocompact_atTop.atTop_mul_atTop tendsto_norm_cocompact_atTop
+
+lemma cocompact_strip.tendsto_sq_im_atTop : Tendsto (fun z : ℂ ↦ z.im^2)
+    (cocompactStrip 0 1) atTop := by
```

Comment:
If the statement doesn't fit on one line, it's better to have all the mathematical content on a single line, like
```suggestion
lemma cocompact_strip.tendsto_sq_im_atTop :
    Tendsto (fun z : ℂ ↦ z.im^2) (cocompactStrip 0 1) atTop := by
```
Same thing below.